tests/e2e/cairo1/test_cairo1_init.py

```diff
@@ -33,7 +33,7 @@ def test_init_with_invalid_name(protostar: ProtostarFixture):
     output = protostar(["init-cairo1", project_name], expect_exit_code=1)
 
     assert (
-        f"Provided project name {project_name} does not match regex ^[a-zA-Z0-9][a-zA-Z0-9_]*$. "
+        f"Provided project name {project_name} does not match the regex ^[a-zA-Z_][0-9a-zA-Z_]*$ or is equal to '_'. "
```

Comment:
I don't think checking the regex explicitly here does us any good really. Would only check for the readable part of the message

Reply:
I don't see a reason not to check it honestly, this way we can make sure that the whole error message is correct.

Reply:
> Provided project name {project_name} in incorrect, project name should start with a letter

Maybe? I agree regex in the error message seems off